### PR TITLE
Mini mirror/multi-monitor bug fixes

### PR DESCRIPTION
**Pull request by SamGondelman**
Fixes several issues relating to the Mini Mirror and multi-monitor displays with different device pixel ratios:
- Mini mirror either rendering at half or twice the correct size on multi monitor setups with different device pixel ratios (fogbugz case 892)
- QML overlays getting messed up when moving between monitors with different device pixel ratios
- In Fullscreen Mirror Mode, mini mirror would show garbage (instead: don't show it)

### CODE REVIEW

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -1459,13 +1459,7 @@ void Application::initializeUi() {
     });
     offscreenUi->resume();
     connect(_window, &MainWindow::windowGeometryChanged, [this](const QRect& r){
-        static qreal oldDevicePixelRatio = 0;
-        qreal devicePixelRatio = getActiveDisplayPlugin()->devicePixelRatio();
-        if (devicePixelRatio != oldDevicePixelRatio) {
```
Comment (SamGondelman) [inline interface/src/Application.cpp]: Correct me if I'm wrong but it seemed like this check was redundant with the check in resizeGL.

**Comment by SamGondelman** [comment]
Testing:

- Connect two monitors with different device pixel ratios.  If they have different device pixel ratios, moving Interface between them should trigger this to be logged: "Device pixel ratio changed, triggering resize"
- Run the default scripts so there are buttons and other overlays around the screen.
(I'm going to refer to the monitor that Interface first starts on as Window 1 and the other one as Window 2)
- Start Interface on Window 1.  Make sure the mini mirror is active.  Move Interface to Window 2.  Close Interface.
- Start Interface on Window 2 (it should start there automatically after closing it).  Move it to Window 1.
- Resize Interface on both windows.
- Try changing the scale resolution (Developer->Render->Scale Resolution)
- At all times, the mini mirror should remain the correct size, and the overlays should remain in the correct places.
- Toggle Fullscreen Mirror mode with 'H' or the View menu items.  The mini mirror should close and reopen when you switch back and forth between modes.

**Comment by hifi-gustavo** [comment]
Build console output for commit: d4cfe6256a93884bf13f6869f1c2192bd6e4c442 available at: https://gist.github.com/c551701b70680a15d6090547da59b01f<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8121.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8121.dmg

**Comment by judasshuffle** [comment]
Testing: using a 4k monitor and a HD monitor

    Connect two monitors with different device pixel ratios. If they have different device pixel ratios, moving Interface between them should trigger this to be logged: "Device pixel ratio changed, triggering resize" 
I got a "window has been resized message" in the log a "06/24 09:50:20] [DEBUG] Offscreen UI resizing to  1703 x 937  with pixel ratio  1"


    Run the default scripts so there are buttons and other overlays around the screen.
I did that .

    (I'm going to refer to the monitor that Interface first starts on as Window 1 and the other one as Window 2)
    Start Interface on Window 1. Make sure the mini mirror is active. Move Interface to Window 2. Close Interface. 
Did that

    Start Interface on Window 2 (it should start there automatically after closing it). Move it to Window 1.
 That worked
    Resize Interface on both windows.
    Try changing the scale resolution (Developer->Render->Scale Resolution)
    At all times, the mini mirror should remain the correct size, and the overlays should remain in the correct places. Changing the scale resolution should still work as it did before.
That worked.
    Toggle Fullscreen Mirror mode with 'H' or the View menu items.
 The mini mirror should close and reopen when you switch back and forth between modes.
The mirror remained but the audio mute buttons disappeared.

**Comment by SamGondelman** [comment]
@judasshuffle What do you mean by "disappeared"?  They should move up a little bit like they do in master, but I never see them disappear.  And it looks like that was the only issue you saw?

**Comment by judasshuffle** [comment]
@SamGondelman https://i.gyazo.com/94a0a10a1c3b99f71ac3497034e642f9.png
on that the bit with the mute sound and face vanished but the mirror remained

**Comment by SamGondelman** [comment]
I've talked with Judas and this seems to be behaving correctly and we can't reproduce any issues, so I'll pass it now and if any bugs become apparent later I'll deal with them.

**Pull request merged**